42 generate plan view

TobiasFitzke01 opened the pull request:
Erstellung der View zum Generieren eines Trainingpplans.
Da das Backend noch nicht so weit ist konnte es noch nicht komplett final getestet werden aber so habt ihr schonmal die Gelegenheit einen Blick drauf zu werfen.

TobiasFitzke01 committed:
Create File for View

TobiasFitzke01 committed:
Erstellen aller notwendigen Files für Plangenerierung und erste Optische Version der Forms

TobiasFitzke01 committed:
Weiterarbeit an Trainingsplan generierung

TobiasFitzke01 committed:
Paar Anpassungen und Rausschmiss von unötigem Code

TobiasFitzke01 committed:
numberOfTraininssession zu int konvertiert

TobiasFitzke01 committed:
Mehr Trainingstage stehen jetzt zur Auswahl

steffenhls committed:
Merge branch 'main' into 42-generate-plan-view

steffenhls commented on tromega/lib/data/generate_plan_http_helper.dart:
```diff
@@ -0,0 +1,24 @@
+class GeneratePlanHttpHelper {
+  final String authority = 'api.fitnessapp.gang-of-fork.de';
+
+  Future<String> postGeneratedPlanPreferences(
```
Ich weiß nicht wie sehr du das mit dem Backend abgesprochen hast, aber es wäre eigentlich schön, wenn du ein [TrainingPlan](https://github.com/projekt-fitnessapp/frontend/blob/main/tromega/lib/data/trainingPlan.dart) wiederbekommst und auch dementsprechend in die (PlanDetailsView)[https://github.com/projekt-fitnessapp/frontend/blob/main/tromega/lib/views/plans/plan_details.dart] mit dem Trainingsplan gehst. Ansonsten ist das so aber auch okay.

TobiasFitzke01 replied:
Plan war erstmal einfach zu sagen der Plan wird im Backend erstellt und dann geh ich einfach zurück auf die Plan Übersicht. Ich weeiß nicht, was sagt ihr, was ist besser, bzw. soll das Backend es erstmal zum laufen bekommen und dann schauen wir oder direkt ansprechen und den Plan ändern?

steffenhls commented on tromega/lib/views/plans/generate_training_view.dart:
```diff
@@ -0,0 +1,44 @@
+        physics: const BouncingScrollPhysics(),
+        children: [
+          const SizedBox(height: 24),
+          GeneratePlanQuestionWidget(
```
Schön :)

steffenhls commented on tromega/lib/views/plans/plan_overview.dart:
```diff
@@ -88,7 +89,9 @@ class _PlanOverviewState extends State<PlanOverview> {
                                                     .style,
-                                                onPressed: () {},
+                                                onPressed: () async {
+                                                  newGeneratedTrainingPlan();
```
Sehr leserlich 👍

steffenhls commented on tromega/lib/widgets/plan/generated_plan_questions_widget.dart:
```diff
@@ -0,0 +1,113 @@
+  late GeneratedPlanPreferences preferences;
+
+  @override
+  void initState() {
+    getUserId();
+    super.initState();
+  }
+
+  getUserId() async {
+    final prefs = await SharedPreferences.getInstance();
+    var userId = prefs.getString("userId");
+    userId ??= "634dad62663403c8063adc41";
+
+    preferences = GeneratedPlanPreferences(userId, 2, "", "");
+  }
```
In der [PlanOverview](https://github.com/projekt-fitnessapp/frontend/blob/main/tromega/lib/views/plans/plan_overview.dart) wird die userId bereits abgefragt, kannst sie theoretisch dort als Variable initialisieren und der View direkt mitgeben, dann kannst du dir die Asynchronität sparen. Trotzdem fine

steffenhls commented on tromega/lib/widgets/plan/generated_plan_questions_widget.dart:
```diff
@@ -0,0 +1,113 @@
+  late List<String> trainingStatusOptions = ["Untrainiert", "Trainiert"];
+
+  String trainingsType = "Mit Maschinen";
+  late List<String> trainingsTypeOptions = ["Mit Maschinen", "Ohne Maschinen"];
```
Sind die Parameter mit Backend abgesprochen?

steffenhls replied:
Gleiche Anmerkungen wie oben

steffenhls commented on tromega/lib/widgets/plan/generated_plan_questions_widget.dart:
```diff
@@ -0,0 +1,113 @@
+  }
+
+  String numberOfTraininssession = "2";
+  late List<String> numberOfTraininssessionOptions = [
```
Warum initalisierst du die mit late, obwohl du die direkt vergibst?

steffenhls replied:
Trainin**g**session

steffenhls commented on tromega/lib/widgets/plan/generated_plan_questions_widget.dart:
```diff
@@ -0,0 +1,113 @@
+    preferences = GeneratedPlanPreferences(userId, 2, "", "");
+  }
+
+  String numberOfTraininssession = "2";
```
Würde das in Folge der Liste als `numberOfTraininsessionOptions[0]` definieren

steffenhls commented on tromega/lib/widgets/plan/generated_plan_questions_widget.dart:
```diff
@@ -0,0 +1,113 @@
+            DropDownWidget(
+              color: color,
+              items: numberOfTraininssessionOptions,
+              currentValue: numberOfTraininssession.toString(),
```
ist doch bereits als String deklariert oder?

steffenhls replied:
Nicht schlecht, aber finde das für die Zahlen bisschen merkwürdig, weil da viel weißer leerer platz ist, aber erstmal denke fine

steffenhls commented on tromega/lib/widgets/plan/generated_plan_questions_widget.dart:
```diff
@@ -0,0 +1,113 @@
+              },
+            ),
+            const SizedBox(height: 16),
+            buildQuestion(text: 'Wie traniert bist du?'),
```
trainiert

steffenhls commented on tromega/lib/widgets/plan/generated_plan_questions_widget.dart:
```diff
@@ -0,0 +1,113 @@
+                this.trainingsStatus = trainingsStatus;
+              },
+            ),
+            const SizedBox(height: 16),
```
Du arbeitest viel mit `SizedBox`. Wenn du sowieso oben und unten Abstand willst, kannst du es gerne in ein Padding aufnehmen.

steffenhls reviewed (approved):
Echt tolle Arbeit, habe ein paar kleinere, teilweise oberflächlichere Anmerkungen gemacht. Falls du iwas nicht nachvollziehen kannst, schreib mir einfach.
Hättest direkt im Frontend durchstarten sollen! Sehr solide 💯

steffenhls commented on tromega/lib/views/plans/generate_training_view.dart:
```diff
@@ -0,0 +1,44 @@
+              generatePlanHttpHelper
+                  .postGeneratedPlanPreferences(generatedPlanPreferences)
+                  .then((value) =>
+                      Navigator.pushNamed(context, '/myWorkoutPlans'));
```
Wenn du die Navigation nicht änderst, bitte entweder wie in der Home View navigieren oder pop nutzen. Gibt sonst keine Navigation Bar

TobiasFitzke01 replied:
Wenn ich da pop mache navigiert er mich zwar zurück aber dann ist noch das widget mit Manuell oder Automatisch offen. Wie meinst du das mit dem navigieren in der home view?

steffenhls replied:
Es gibt auch ein popUntil. Sollte damit eigentlich gehen.

TobiasFitzke01 committed:
Navigation zurück zur planübersicht angepasst

Schreifehler angepasst
SizedBoxes durch padding ersetzt
Anpsssung der Dropdownauswahl definition

Veerschieben der verwendeten Widgets in shared folder, da Rebekka und ich die selben verwenden. sie wird es bei sich auf die shared files anpassen und ihre alten files löschen

steffenhls commented on tromega/lib/views/plans/generate_training_view.dart:
```diff
@@ -33,8 +34,11 @@ class _GeneratePlanViewState extends State<GeneratePlanView> {
+                  .then((value) => Navigator.push(
+                      context,
+                      MaterialPageRoute(
+                        builder: (context) => App(currentIndex: 1),
```
👍

steffenhls commented on tromega/lib/widgets/plan/generated_plan_questions_widget.dart:
```diff
@@ -34,21 +34,21 @@ class _SecondGeneratePlanQuestionWidget
     "6",
     "7"
   ];
+  late String numberOfTrainingssession = numberOfTrainingssessionOptions[0];
```
müssen die late sein?

steffenhls commented on tromega/lib/widgets/plan/generated_plan_questions_widget.dart:
```diff
@@ -92,14 +87,13 @@ class _SecondGeneratePlanQuestionWidget
                 text: 'Plan generieren',
                 onClick: () {
                   setState(() {
                     preferences.numberOfTraininssession =
```
bitte auch über generatedPlanPreferences.dart gucken

steffenhls commented on tromega/lib/widgets/shared/data_widget.dart:
```diff
@@ -0,0 +1,19 @@
+import 'package:flutter/material.dart';
+
+Widget buildQuestion({required String text}) => Padding(
+      padding: const EdgeInsets.only(left: 10, bottom: 16),
```
👍

steffenhls reviewed (commented):
bitte nochmal in generatedPlanPreferences.dart über das Wort Training gucken. Kannst danach eigenständige mergen

TobiasFitzke01 committed:
fix typos
